agent/engine/docker_task_engine.go

```diff
@@ -447,8 +448,19 @@ func (engine *DockerTaskEngine) AddTask(task *api.Task) error {
 
 	existingTask, exists := engine.state.TaskByArn(task.Arn)
 	if !exists {
+		// This will update the container desired status
+		task.UpdateDesiredStatus()
+
 		engine.state.AddTask(task)
-		engine.startTask(task)
+		if dependencygraph.ValidDependencies(task) {
+			engine.startTask(task)
```

Comment:
I think @aaithal means get rid of the `else` word but keeps the statement inside the else, see my updates.